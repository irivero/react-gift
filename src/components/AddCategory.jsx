import { useState } from "react"

export const AddCategory = ({onNewValue}) => {
    const [category, setcategory] = useState('')

    const onChange = ({target}) => {
        setcategory(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        if(category.trim().length === 0) return;
        onNewValue(category.trim());        
        setcategory('');
    }

    return (
        <form onSubmit={onSubmit}>
            <input  type= "text" 
                    placeholder="Find category"
                    value={category} 
                    onChange={ onChange }
            />
        </form>
    )
}