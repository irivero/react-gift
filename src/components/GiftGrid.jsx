import { GridItem } from "./GridItem";
import { useFetchGifts } from "../hooks/useFetchGifts";

export const GiftGrid = ({category}) => {

    const {images, isLoading} = useFetchGifts(category);

    return(       
            <>
                <h3>{category}</h3>
                {
                    isLoading 
                    && (<h2>Is loading...</h2>) 
                    
                }
                
                <div className="card-grid">
                    {
                        images.map((img)=> <GridItem key={img.id} {...img}/>)
                    }
                </div>
            </>       

    )

}