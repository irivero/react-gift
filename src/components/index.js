
export * from './AddCategory';
export * from '../GiftExpertApp';
export * from './GiftGrid';
export * from './GridItem';


