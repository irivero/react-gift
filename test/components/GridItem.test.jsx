import { render , screen} from "@testing-library/react"
import { GridItem } from "../../src/components/GridItem"

describe('GridItem Tests', () => { 
    const testTitle = 'Gumball';
    const testUrl = 'https://pictures.com/gumball/gumball_playing.jpg';

    test('should match with the snapshot', () => {         
     const {container} =   render(<GridItem title ={testTitle} url ={testUrl} />)
     expect(container).toMatchSnapshot()
    
    })

    test('should show the image with the ALT', () => { 
        
        render(<GridItem title = {testTitle} url = {testUrl} />)
        screen.debug()

        const {src, alt} = screen.getByRole('img');
        expect (src).toBe(testUrl);
        expect(alt).toBe(testTitle);
     })

     test('should display the title', () => { 
        render(<GridItem title = {testTitle} url = {testUrl} />)
       
        expect(screen.getByText(testTitle)).toBeTruthy();
      })
 })
