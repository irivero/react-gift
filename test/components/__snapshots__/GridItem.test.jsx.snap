// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem Tests should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gumball"
      src="https://pictures.com/gumball/gumball_playing.jpg"
    />
    <p
      data-test-id="title"
    >
      Gumball
    </p>
  </div>
</div>
`;
